.github/workflows/_deploy_kafka_test.yml

```diff
@@ -0,0 +1,198 @@
+      sidecar_registry:
+        required: true
+        description: "The registry to get the image from"
+        type: string
+      key-release-image: 
+        required: true 
+        description: "The image of the SKR sidecar to use"
+        type: string
+jobs:
+  setup-workload-identity:
+    name: Deploy Kafka Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: main
+    
+      - name: Checkout tools repo 
+        uses: actions/checkout@v4
+        with:
+          repository: stevendongatmsft/confidential-container-demos
+          path: confidential-container-demos
+
+      - name: Log into Azure
+        run: |
+          az login --service-principal \
+            --username ${{ secrets.AZURE_SERVICE_PRINCIPAL_APP_ID }} \
+            --password ${{ secrets.AZURE_SERVICE_PRINCIPAL_PASSWORD }} \
+            --tenant ${{ secrets.AZURE_SERVICE_PRINCIPAL_TENANT }}
+            
+      - name: Install Dependencies 
+        id: install-dependencies 
+        run: |
+            curl -L https://github.com/a8m/envsubst/releases/download/v1.2.0/envsubst-`uname -s`-`uname -m` -o envsubst
+            chmod +x envsubst
+            sudo mv envsubst /usr/local/bin
+
+            result=$(az extension list -o table  2>&1 || true)
+            if [[ $result == *"confcom"* ]]; then
+              echo "confcom already installed, upgrading confcom version."
+              az extension update --name confcom
+            else
+              echo "confcom extension not found. Installing confcom..."
+              az extension add --name confcom
+            fi
+
+            curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+            chmod +x kubectl
+            sudo mv kubectl /usr/local/bin
+            kubectl version 2>&1 || true
+
+      - name: Run Workload
+        env:
+          SkrClientKID : ${{ inputs.skr-client-kid }}
+          SkrClientAKVEndpoint: ${{ inputs.skr-client-akv-endpoint }}
+          SkrClientMAAEndpoint: ${{ inputs.skr-client-maa-endpoint }}
+          TOPIC: ${{ inputs.skr-kafka-demo-topic }}
+          CLUSTER_NAME: ${{ inputs.cluster-name }}
+          RESOURCE_GROUP: ${{ vars.RESOURCE_GROUP }} 
+          SIDECAR_IMAGE: ${{ format('{0}/{1}', inputs.sidecar-registry, inputs.key-release-image) }}
+        id: run-workload
+        run: | 
+          az aks get-credentials --name $CLUSTER_NAME --resource-group $RESOURCE_GROUP  --overwrite-existing
+          cd confidential-container-demos/kafka
+
+          SkrClientAKVEndpoint=${SkrClientAKVEndpoint#"https://"}
+          SkrClientAKVEndpoint=${SkrClientAKVEndpoint#"http://"}
+          export SkrClientAKVEndpoint=${SkrClientAKVEndpoint%%/*}
+
+          echo "SkrClientKID value is: $SkrClientKID"
+          export SkrClientKID=$SkrClientKID
+          echo "SkrClientMAAEndpoint value is: $SkrClientMAAEndpoint"
```

Comment:
I don't think these echos do much and since we should be using Github Secrets for their values, we wouldn't want them printed in a pipeline anyway.

Reply:
ok I can remove these echos